ontology/yaml/resources/HVAC/entity_types/ABSTRACT.yaml

```diff
@@ -6791,6 +6791,15 @@ OAQM:
   implements:
   - MONITORING
 
+UDEC:
+  description: "Up and down extension control for shades."
+  is_abstract: true
+  uses:
+  - up_shade_command
+  - down_shade_command
+  opt_uses:
+  - shade_extent_percentage_sensor
```

Comment:
Does this device have `shade_extent_percentage_command` ? It seems odd to have the up/down commands without a position command; for example, `shade_extent_percentage_command` is essentially the position setpoint that would tell the up/down commands when to start or stop.

Reply:
I agree with you it's odd. However, there is no position command. The only way to know when starting and stopping the up and down command it's by calculating the duration of the command according to the current position and the shade speed. For exemple, if the controller want to go from 80% to 20% and it's known the shade go up at the speed of 10% per second, the up command will be apply 6 seconds.

Reply:
Ok thank you for clarifying.